types/openid-client/index.d.ts

```diff
@@ -167,13 +172,27 @@ export class Client {
      * @param extras Additional revoke options
      */
     revoke(token: string, tokenTypeHint: string, extras?: RevokeRequestOptions): Promise<void>;
+
+    /**
+     * Refresh your active token
+     * @param refreshToken The refresh token
+     * @param opts Additional options
+     */
+    refresh(refreshToken: string, opts?: RefreshRequestOptions): Promise<TokenSet>;
 }
 
-export class TokenSet {
+type PartialRecord<K extends keyof any, T> = {
+    [P in K]?: T;
+};
+
+export class TokenSet implements PartialRecord<string, any> {
     readonly access_token?: string;
     readonly token_type?: string;
     readonly id_token?: string;
     readonly refresh_token?: string;
+    readonly expires_at?: number;
+    readonly expires_in?: number;
+    readonly [key: string]: any;
```

Comment:
No, please don't add an `any` result type. That's not type-safe.

Either just remove the indexer, or use `unknown` (see the other indexers). 

Please also remove the `implements PartialRecord<string, any>` and the type. It's unnecessary.